sceptre/helpers.py

```diff
@@ -68,6 +68,28 @@ def func_on_instance(key):
     return attr
 
 
+Container = Union[list, dict]
+Key = Union[str, int]
+
+
+def delete_keys_from_containers(keys_to_delete: List[Tuple[Container, Key]]):
```

Comment:
This is mostly just extracted out of the ResolvableProperty class and put here. I needed to also use this in the CustomYamlTagBase class.

Reply:
I'm glad that this has been extracted but I'm a bit queasy about this function mutating collections it doesn't own. I got side-tracked by work but I feel this could be simplified (in the future) using list comprehensions and/or `filter`. But then I wasn't able to dig enough into how it's used to propose a refactor.

Reply:
A couple responses:
1. It _does_ own them. That is why there is the cloning functionality of resolvers. The only collections that are mutated are ones that are effectively deep-copied when they are assigned to the stack.
2. This is generally the way resolvers have always worked. The basic idea is that it prevents us from ever needing to resolve the same resolver twice for the same stack. This can be important, since the resolvable attribute might be accessed a _number_ of times over the course of execution.
3. Once we've resolved the resolver, there's no sense in keeping it around; its purpose has been fulfilled.